server/src/server.rs

```diff
@@ -47,14 +47,14 @@ impl CubebDeviceCollectionManager {
         server: &Rc<RefCell<DeviceCollectionChangeCallback>>,
         devtype: cubeb::DeviceType,
     ) -> cubeb::Result<()> {
-        let mut servers = self.servers.lock().unwrap();
-        if servers.is_empty() {
-            self.internal_register(context, true)?;
-        }
         server.borrow_mut().devtype.insert(devtype);
+        let mut servers = self.servers.lock().unwrap();
         if !servers.iter().any(|s| Rc::ptr_eq(s, server)) {
             servers.push(server.clone());
         }
+        if servers.len() == 1 {
+            self.internal_register(context, true)?;
+        }
```

Comment:
I don't think I understand why `servers.is_empty()` is changed to `servers.len() == 1`? Would you mind explaining it a bit?

Reply:
Thanks for querying this, the logic is slightly wrong now.  I'll fix it up.